upup/pkg/fi/executor.go

```diff
@@ -102,13 +103,31 @@ func (e *executor) RunTasks(taskMap map[string]Task, maxTaskDuration time.Durati
 
 		taskErrors := e.forkJoin(tasks)
 		var errors []error
+
 		for i, err := range taskErrors {
 			ts := tasks[i]
+			var lifecycle *Lifecycle
+			if hl, ok := ts.task.(HasLifecycle); ok {
+				lifecycle = hl.GetLifecycle()
+			}
+
 			if err != nil {
 				remaining := time.Second * time.Duration(int(ts.deadline.Sub(time.Now()).Seconds()))
-				glog.Warningf("error running task %q (%v remaining to succeed): %v", ts.key, remaining, err)
 				errors = append(errors, err)
 				ts.lastError = err
+				if lifecycle == nil {
+					glog.Warningf("error running task %q (%v remaining to succeed): %v", ts.key, remaining, err)
```

Comment:
This check is alredy done in `default` on the switch below

Reply:
Cool was not certain

Reply:
hmm, actually you might get a nil pointer cause its *lifecycle :P

Reply:
Arg let me fix that above then